### PR TITLE
Improved context when exiting with error

### PR DESCRIPTION
**Pull request by superlinkx**
fix: added context to every exit()
chore: refactored common connectionTest() and newAzureClient() calls into its own reusable function
chore: fixed a few linter issues, including two unhandled error cases
chore: added break if currentTask errors and updated log message
chore: fixed several instances of bad json tags that were flagged by go vet

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -275,3 +277,16 @@ func updateClient(ctx context.Context, bheUrl url.URL, bheClient *http.Client) e
 		}
 	}
 }
+
+func connectAndCreateClient() client.AzureClient {
```
Comment (ddlees) [inline cmd/start.go]: Tasty refactor. Can we move this to `utils.go`?

File: cmd/start.go [hunk 2]
```diff
@@ -128,7 +124,10 @@ func start(ctx context.Context) {
 
 								// Notify BHE instance of task start
 								currentTask = &executableTasks[0]
-								startTask(ctx, *bheInstance, bheClient, currentTask.Id)
+								if err := startTask(ctx, *bheInstance, bheClient, currentTask.Id); err != nil {
+									log.Error(err, "failed to start task")
+								}
```
Comment (ddlees) [inline cmd/start.go:130]: If we fail to start the task then we should probably unset currenttask and try again after the next heartbeat

Comment (superlinkx) [reply cmd/start.go:130]: Yeah, I didn't know for sure what the best error state would be here, so just logged for now. More than happy to add this behavior while we're here

File: cmd/start.go (hunk 1 above)
Comment (superlinkx) [reply cmd/start.go]: Yeah, that's easy

**Review by ddlees — APPROVED** [review]

**Pull request merged**